ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,365 @@ def decorated(*args, **kwargs):
+@app.route("/job/create", methods=["POST"])
+@token_required
+def job_create_endpoint(current_user):
+    """
+    API to create a job using the existing template.
+
+    IMPORTANT: template_id NEEDS to be provided in the request body. job_name can also be provided in the request body.
+    """
+
+    # Imports
+    from GangaCore.GPI import templates, Job
+
+    # Store request data
+    template_id: int = request.form.get("template_id")
+    job_name: str = request.form.get("job_name")
+
+    # template_id existence check
+    if template_id is None:
+        return jsonify({"success": False,
+                        "message": "Template ID not provided in the request data"}), 400
+
+    # Integer check
+    try:
+        template_id = int(template_id)
+    except ValueError:
+        return jsonify({"success": False,
+                        "message": "Template ID provided in not an Integer"}), 400
+
+    # Template existence check
+    if int(template_id) not in templates.ids():
+        return jsonify({"success": False,
+                        "message": "Template with ID {} does not exist".format(template_id)}), 400
+
+    # Create job using template
+    try:
+        j = Job(templates[int(template_id)])
+        if job_name is not None:
+            j.name = job_name
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)})
+
+    return jsonify({"success": True,
+                    "message": "Job with ID {} created successfully using the template ID {}".format(j.id,
+                                                                                                     template_id)})
+
+
+# Perform certain action on the Job - PUT Method
+@app.route("/job/<int:job_id>/<action>", methods=["PUT"])
+@token_required
+def job_action_endpoint(current_user, job_id: int, action: str):
+    """
+    Given the job_id and action in the endpoint, perform the action on the job.
+
+    Supported actions: do_auto_resubmit, name, parallel_submit, copy, kill, force_status, resubmit, runPostProcessors, submit
+
+    Important Info:
+    1) do_auto_resubmit: User needs to provide 'do_auto_resubmit' field and it's value in the request body.
+    2) name: User needs to provide 'name' field and it's value in the request body.
+    3) parallel_submit: User needs to provide 'parallel_submit' field and it's value in the request body.
+    4) force_status: User needs to provide 'force_status' field and it's value in the request body.
+
+    :param job_id: int
+    :param action: str
+    :param attribute: str
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+
+    # Job existence check
+    if job_id not in list(jobs.ids()):
+        return jsonify({"success": False, "message": "Job with ID {} does not exist".format(job_id)}), 400
+
+    supported_actions = ["do_auto_resubmit", "name", "parallel_submit", "copy", "kill", "force_status", "resubmit",
+                         "runPostProcessors", "submit"]
```

Comment:
This seems to replicate information that is already in the Job class, see https://github.com/ganga-devs/ganga/blob/92c6ac046b8b323d5428434289ee430c7dd2375a/ganga/GangaCore/GPIDev/Lib/Job/Job.py#L233. Can't we just take it from there to avoid having to maintain the same information in two different places?

Reply:
The supported_actions list is just for validating that the action the user wants to perform is actually allowed or not. 

Eg: Suppose a user tries to change the name of the job (with ID 13), then they will make a `PUT` request at the endpoint `/job/13/name`. The `name` part of the endpoint is checked if it exists in the supported_actions list so that it is allowed. Which in this case of is allowed. But suppose, if the user uses the endpoint `/job/13/newname`, here the `newname` action is not allowed, and the user is returned with the HTTP code 400.